packages/data/src/navigation/actions.js

```diff
@@ -16,6 +16,13 @@ export function setMenuItems( menuItems ) {
 	};
 }
 
+export function setPersistedQuery( persistedQuery ) {
```

Comment:
I would love to see these actions more "event" driven.  We haven't done a great job of this in the past when creating data stores, but I have some prior art in the nav favorties data store.

I'm thinking the event driving the persisted query might be `PAGE_LOADED` or `HISTORY_CHANGED` or a combo potentially.